### ISSUE TITLE
[docs] Add a "Getting Started with optimization" tutorial

### ISSUE THREAD

**Issue by jd-foster** [jump-dev/JuMP.jl#2859]
Would you accept a PR for the following?: a page in the docs in the Tutorial section called "Getting Started with optimization" that gives an overview of the math. formulation of linear, conic, quadratic and nonlinear programs? At the moment, the descriptions are scattered between each subsection, and may be hard for a learner to fit their problem type into the right category. An extension to this would be a "decision tree" format to help direct classification.

**Comment by jd-foster** [comment]
Just realised I should probably have just added this under #2348 ...

**Comment by odow** [comment]
Yeah I've been thinking along these lines. (It's why I added the various introductions.)

We currently have a mix of:
 * https://jump.dev/JuMP.jl/dev/should_i_use/
 * https://jump.dev/JuMP.jl/dev/background/algebraic_modeling_languages/
 * https://jump.dev/JuMP.jl/dev/tutorials/linear/introduction/
 * https://jump.dev/JuMP.jl/dev/tutorials/conic/introduction/
 * https://jump.dev/JuMP.jl/dev/tutorials/nonlinear/introduction/

Summarizing all of this into a single "getting started" tutorial to help people identify what types of problems JuMP can solve, and what type of problem they have/what solver they should be looking for would be useful.

I wondered about going for some of the early AMPL chapters, like "what is a decision variable/objective/constraint", but we need to decide what should go in the JuMP documentation and what should be left out.

If https://jump.dev/JuMP.jl/dev/JuMP.pdf is to become "JuMP, the book" at some point, I think we need an intro to modelling/optimization chapter covering this sort of stuff, even if only briefly.

> Just realised I should probably have just added this under

I think this deserves a separate issue. It's probably the next biggest issue that new users face with the tutorials.

**Comment by jd-foster** [comment]
> I wondered about going for some of the early AMPL chapters, like "what is a decision variable/objective/constraint", but we need to decide what should go in the JuMP documentation and what should be left out.

I've read the AMPL book extensively, and I like how it's very example+language driven in its approach. I can see the scope creep for this proposal, but the primary purpose is would be to let the reader 'fall through' the problem types and identify the most specific type of program that covers their formulation. There is a whole separate issue of teaching how to identify and perform re-formulation.

**Comment by odow** [comment]
One thing that always surprises me about the AMPL book (or at least the 2nd edition that I have) is that it is primarily LP. It's not until Chapter 18 that we get nonlinear, Chapter 19 introduces complementarity problems (!), and then Chapter 20 introduces integer problems.

> I can see the scope creep for this proposal

Yes. But having said that, great introductory material will only help more people use JuMP and make it easier to teach in classes.

**Comment by jd-foster** [comment]
I've put an initial intro down [here](https://github.com/jd-foster/JuMP.jl/blob/5dbbf27ce86b5812c177945ba75262dbf2cfaec6/docs/src/tutorials/getting_started/getting_started_with_optimization.jl).

I would appreciate initial comments on whether this is hitting the right tone and reader level we're aiming for. The wording is still quite rough.

**Comment by odow** [comment]
The introduction in Jorge's Numerical Optimization book is quite good. The gist is:

 * We want to optimize decision in the real world
 * We need to _model_ that world in mathematics
 * These models have decision variables, constraints, and an objective
 * Concrete example (e.g., knapsack)
 * Generic mathematical formulation (e.g., the MOI f-in-S)
 * Then break down the world. What structures of the functions and sets make these problems easy or hard?
   * Linear vs nonlinear
   * Convex vs nonconvex
   * Continuous vs discrete
   * Constrained vs unconstrained
   * Global vs local
   * Stochastic vs deterministic
   * Structure vs black-box

My guess is that the best way forward is to open a PR, but it's probably going to take quite a lot of work-shopping to get right.

**Comment by odow** [comment]
Since we haven't made progress on this, I'm going to close in favor of the generic #2348. We do have to be careful about the scope of the JuMP documentation. It's not a textbook for LP101.

**Comment by jd-foster** [comment]
Sounds good. I think priority should be given to readable tutorials that embed most content.
